app/src/main/java/it/quip/android/model/User.java

```diff
@@ -142,18 +110,20 @@ public static User getUserForSession() {
         }
 
         if (null == currentUser) {
-            currentUser = new User();
-            currentUser.setFacebookId(facebookId);
-            currentUser.saveInBackground();
+            try {
+                FacebookClient.createNewUser();
```

Comment:
If it ever reaches this point, its because no Parse User object was found with the corresponding Facebook ID. It is possible that the previous check failed due to network issues, in which case we'd have to come up with an atomic check and create method that would only create the account if the Facebook ID didn't already exist. That's outside of the MVP though.